.github/workflows/continuous-integration.yml

```diff
@@ -23,7 +23,7 @@ jobs:
     env:
       # Core variables:
       FC: ${{ matrix.fortran-compiler }}
-      FCFLAGS: "-ffree-line-length-none -m64 -std=f2008 -march=native -fbounds-check -finit-real=nan -g -DRTE_USE_CBOOL -DRTE_USE_${{ matrix.fpmodel }}"
+      FCFLAGS: "-ffree-line-length-none -m64 -std=f2008 -march=native -fbounds-check -fmodule-private -fimplicit-none -finit-real=nan -g -DRTE_USE_CBOOL -DRTE_USE_${{ matrix.fpmodel }}"
```

Comment:
@RobertPincus `-fmodule-private` helps in detecting visibility errors like in the case of the `zero_array` interface. Without this flag, the compiler assumes that `zero_array` is public and the compilation succeeds. With this flag, the compilation fails because the compiler assumes that `zero_array` is private and, therefore, not available to other modules. I noticed the problem because we compile with `-fmodule-private` in our ICON CI and decided that it would be a good idea to have the flag here as well.

As for `-fimplicit-none`, the idea is similar: if there is a variable in the code that does not have an explicitly set type, the compilation fails.

Reply:
Thanks. All the codes should have `implicit none` but better to add extra guard rails.